scripts/sync_versions.sh

```diff
@@ -0,0 +1,12 @@
+#! /bin/bash
+
+# Get current version of quil-rs
+PACKAGE_VERSION=$(cargo metadata --format-version 1 --no-deps | grep -o '"name":"quil-rs","version":"[^"]*' | cut -d '"' -f 8)
+
+# Write it to `quil-py`s Cargo.toml
+# `awk -i inplace` would work on linux, but not on macos. This command is compatible with both.
+awk -v ver="$PACKAGE_VERSION" '/^quil-rs = /{$0="quil-rs = { path = \"../quil-rs\", version = \"" ver "\" }"}1' quil-py/Cargo.toml > quil-py/Cargo.toml.tmp
+mv quil-py/Cargo.toml.tmp quil-py/Cargo.toml
+
+# Validate that the new version of Cargo.toml is valid
+cargo check --manifest-path=quil-py/Cargo.toml
```

Comment:
This is how I'm planning to keep the version of `quil-rs` specified in `quil-py` in sync.

`knope` will call this script after bumping the version of `quil-rs` to update the version spec in `quil-py`. The `cargo check` here returns an error for safety, just in case the awk command mangles the version number.

These are three cases I'm considering:

## A globally scoped release triggers version bumps of both `quil-rs` and `quil-py`

Everything gets version bumped, release tags get created, and the release workflows for both kick off.

## A `quil-py` scoped release triggers a version bump of just `quil-py`

Here, the script is effectively a no-op, this script will run and overwrite the spec with the same version, and the release will proceed as normal.

## A `quil-rs` scoped release triggers a version bump of just `quil-rs`

Here, `quil-rs` gets bumped and so does it's version in `quil-py`'s manifest, but the latter doesn't get released. I don't see this as a feature, since if the release is correctly scoped, the new release of `quil-rs` shouldn't need to trigger a new release of `quil-py`. If it will affect `quil-py`, the release should be scoped to both.